src/python/module/smartredis/dataset_utils.py

```diff
@@ -41,10 +49,9 @@
 
 # ----helper decorators -----
 
-
-def _requires_xarray(fn):
+def _requires_xarray(fn: t.Callable[_PR, _RT])->TFunc:
```

Comment:
Looks like the generic `_PR` and `_RT` are not be followed into the return val here bc of the type allias (similar problem to what we were seeing yesterday)
```py
@_requires_xarray
def dummy(a: int, b: str) -> float:
    ...
reveal_type(dummy)  # reveals `(*Any, **Any) -> Any`
``` 
could we replace this with
```py
def _requires_xarray(fn: t.Callable[_PR, _RT]) -> t.Callable[_PR, _RT]:
```
mypy seems to be happier with that (at least for my little demo fn above)

Reply:
Done here and in util.py